### PR TITLE
Fix broken compiler visibility presets

### LINKED ISSUE #963: C++ is not C
In https://github.com/OpenSCAP/openscap/blob/master/src/CMakeLists.txt#L22
we set  `CXX_VISIBILITY_PRESET hidden`, which is then used on all our objects.

`CXX` means C++, but our code is written in C. Why is it there? Should it be removed?

### PR DESCRIPTION
**Pull request by jan-cerny**
The main problem is that C is not C++, therefore the symbol visibility
setting for C++ is not applicable to C. This caused that many symbols
which must be private were actually exported in libopenscap.so and they
were publicly visible. I have discovered this implication of C++
not being C by using `nm` command and analysing `nm` output carefully.
Fixes #963.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -441,9 +441,13 @@ include_directories(
...
 )
 
+# Honor visibility properties for all target types
+# Run "cmake --help-policy CMP0063" for policy details
+cmake_policy(SET CMP0063 NEW)
+
 function(set_oscap_generic_properties TARGET_OBJECT)
 	set_target_properties(${TARGET_OBJECT} PROPERTIES
-		CXX_VISIBILITY_PRESET hidden    # Make global variables and functions HIDDEN by default.
+		C_VISIBILITY_PRESET hidden    # Make global variables and functions HIDDEN by default.
```
Comment (cipherboy) [inline CMakeLists.txt]: Might as well add in those two spaces...

Comment (mpreisler) [reply CMakeLists.txt]: we want both IMO because dpkg probe is in C++

**Comment by cipherboy** [comment]
@jan-cerny In addition to @mpreisler's comment, it looks like the Jenkins failure is legitimate:

```python
32/87 Test #33: bindings/test_python.sh .............................................***Failed    0.27 sec
+ python_import                                               
TEST: python_import
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/jenkins/workspace/openscap-master-pull-requests/build/swig/python3/openscap_api.py", line 49, in <module>
    OSCAP = _import_helper()
  File "/home/jenkins/workspace/openscap-master-pull-requests/build/swig/python3/openscap_api.py", line 45, in _import_helper
    _mod = imp.load_module('_openscap_py', fp, pathname, description)
  File "/usr/lib64/python3.6/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/usr/lib64/python3.6/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: /home/jenkins/workspace/openscap-master-pull-requests/build/swig/python3/_openscap_py.so: undefined symbol: OSCAP_LANG_DEFAULT
RESULT: FAILED
```

**Comment by jan-cerny** [comment]
So that's pretty unpleasant. I'll try to resolve it.

File: CMakeLists.txt (hunk 1 above)
Comment (jan-cerny) [reply CMakeLists.txt]: OK

**Comment by cipherboy** [comment]
@jan-cerny It looks like it might actually be fairly easy:

```
src/common/text.c:44:const char * const OSCAP_LANG_DEFAULT    = "en-US";
```

I'm guessing it isn't exported / marked public :)

**Comment by jan-cerny** [comment]
I have set both C++ and C visibility due to dpkg probe and I have exported remaining symbols from public API.

The second commit (438d977) fixes problem of Python symbols. These symbols are supposed to be a part of pubilc API, but they have been forgotten to be labeled by OSCAP_API macro. Due to coincidens with Issue #963 this problem was not visible. After Issue #936 has been fixed this issue poped by by breaking our upstream tests of SWIG Python 3 Bindings.

**Comment by cipherboy** [comment]
Thanks @jan-cerny! :)

**Pull request merged**